Improve our e2e tests DX (#2836)
wasp-lang/wasp: waspc/README.md

   - NOTE: You will need to install `ghcid` globally first. You can do it with `cabal install ghcid`.
3. Do a change in the codebase (most often in `src/` or `cli/src/` or `data/`), and update tests if that makes sense (see [Test](#tests)).
   Fix any errors shown by HLS/`ghcid`.

>    Rinse and repeat. If you're an internal team member, postpone updating e2e tests until approval (see [here](#note-for-team-members)).

Reviewer: How do you know I didn't mean to say "posptpone?"

### Note for team members

> Don't update e2e tests until your PR is approved. Running them and accepting changes prematurely can slow down the review process, create noise for reviewers, and makes it less likely that you'll thoroughly check the final diffs if changes are made during review iterations. **Carefully review the diffs before accepting them** (see [End-to-end (e2e) tests](#end-to-end-e2e-tests) for more details).

Reviewer: Let's make it more authoritative and add spacing for emphasis.
I also corrected a minor mistake (e.g., review process -> UI)

```suggestion
Don not update e2e tests until your PR is approved.

Running the e2e tests and accepting them prematurely:
- Slows down the UI.
- Creates noise for the reviewers.
- Makes it less likely that you'll thoroughly check the final diffs after all review iterations.

**Carefully review e2e test diffs before accepting them** (see [End-to-end (e2e) tests](#end-to-end-e2e-tests) for more details).
```

Besides unit tests, we have e2e tests that run `waspc` on a couple of prepared projects, check that they successfully run, and also compare generated code with the expected generated code (golden output).

This means that when you make a change in your code that modifies the generated code, e2e tests will fail while showing a diff between the new generated code and the previous (golden) one.

> This gives you an opportunity to observe these differences and ensure that they are intentional and that you are satisfied with them. **It is the PR author's (or reviewers if its an outside contribution) responsibility to carefully review these diffs.** Do not blindly accept changes, ensure they align with your intended modifications. If you notice something unexpected or weird, you have an opportunity to fix it.

Reviewer: ```suggestion
This gives you an opportunity to observe these differences and ensure that they are intentional and that you are satisfied with them. **It is the PR author's (or the reviewers for outside contributions) responsibility to carefully review these diffs.** Do not blindly accept changes, ensure they align with your intended modifications. If you notice something unexpected or weird, you have an opportunity to fix it.
```
